File: Java_ChildrenCompetitionSockets/CompetitionPersistence/src/main/java/competition/persistence/ITestRepository.java
package competition.persistence;


//public interface ITestRepository<ID, T extends HasId<ID>> extends IRepository<ID, T>{
//    Iterable<T> findAllTestsForParticipant(int id);
//}

import competition.model.Test;

public interface ITestRepository extends IRepository<Integer, Test>{
    Iterable<Test> findAllTestsForParticipant(int id);
}

File: CSharp_ChildrenCompetitionSockets/CSharp_ChildrenCompetitionSockets/persistence/ITestRepository.cs
using System.Collections.Generic;
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.repository
{
    public interface ITestRepository : IRepository<int, Test>
    {
        IEnumerable<Test> findAllTestsForParticipant(int id);
    }
}